src/movement/maestro.hpp

```diff
@@ -0,0 +1,342 @@
+                if (now > nextReset)
+                {
+                    ROS_INFO_STREAM("Sending thruster reset signal.");
+                    for (int thruster = 0; thruster < thrusters; ++thruster)
+                    {
+                        if (parseNormalized(0, command[thruster*2+4],
+                                command[thruster*2+3]))
+                        {
+                            ROS_DEBUG_STREAM("Parse Normalized encountered
+                                    abnormal thruster speed.");
+                            return -1;
+                        }
+                    }
+
+                    if (port->Write(command, sizeof(command)) !=
+                            sizeof(command))
+                    {
+                        ROS_DEBUG_STREAM("Serial port failed to write entire
+                                command.");
+                        return -1;
+                    }
+
+                    nextReset = ros::Time::now() + ros::Duration(reset_delay);
+
+                    /*
+                     * Sleep for 185ms (9 50Hz cycles + 1/4 of one cycle) to
+                     * ensure that the zero pulse has propogated to the ESC.
+                     * Experimental tests have found this value. Any value less
+                     * is insufficient and will cause ESC malfuctions.
+                     */
+                    usleep(185000);
+                }
+
+                /*
+                 * Send thruster commands down the to the Maestro.
+                 */
+                for (int thruster = 0; thruster < speeds.size(); ++thruster)
+                {
+                    double speed = speeds[thruster];
+                    if (speed < -1 || speed > 1)
+                    {
+                        ROS_DEBUG_STREAM("Thruster speed out of range.");
+                        return -1;
+                    }
+
+                    if (speed > max_speed)
+                    {
+                        ROS_DEBUG_STREAM("Software-limiting thruster speed.");
+                        speed = max_speed;
+                    }
+                    if (speed < -1*max_speed)
+                    {
+                        ROS_DEBUG_STREAM("Software-limiting thruster reverse
+                                speed.");
+                        speed = -1*max_speed;
+                    }
+                    if (parseNormalized(speed, command[2*thruster+4],
+                            command[2*thruster+3]))
+                    {
+                        ROS_DEBUG_STREAM("Parse Normalized encountered abnormal
+                                thruster speed.");
+                        return -1;
+                    }
+                }
+
+                /*
+                 * Write the data down the serial port.
+                 */
+                return (port->Write(command, sizeof(command)) !=
+                        sizeof(command));
+            }
+
+        private:
```

Comment:
Declaring member variables at the top of a header file is easier to read. That way, the reader knows what the class contains without having to trudge through the rest of the file